src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.FormatDocument.cs

```diff
@@ -81,51 +102,49 @@ private bool FormatText(ITextSnapshot currentSnapShot, ITextView textView, Comma
                 return false;
             }
 
-            using (context.WaitContext.AddScope(allowCancellation: true, EditorFeaturesResources.Formatting_document))
-            {
-                Format(textView, document, null, context.WaitContext.UserCancellationToken);
-            }
-
+            Format(textView, document, null, cancellationToken);
             return true;
         }
 
-        private bool ApplyOneCodeFix(ITextSnapshot currentSnapShot, CommandExecutionContext context, out bool hasMoreCodeFix)
+        private Document ApplyCodeFixes(Document document, CancellationToken cancellationToken)
         {
-            hasMoreCodeFix = false;
-            var cancellationToken = context.WaitContext.UserCancellationToken;
-
-            // document needs to be retrieved again after each change, otherwise it does not work
-            var document = currentSnapShot.GetOpenDocumentInCurrentContextWithChanges();
-            if (document == null)
+            var fixAllService = document.Project.Solution.Workspace.Services.GetService<IFixAllGetFixesService>();
+
+            var dummy = new ProgressTracker();
+            foreach (var diagnosticId in new string[] {
+                IDEDiagnosticIds.AddBracesDiagnosticId,
```

Comment:
@kuhlenh I think I added all P1 you said except expression body converting. do you want that one as well for demo? more diagnostics more work get slower. so if we don't need it, leaving out will be better :)